pkg/util/port/portlist/poller_linux.go

```diff
@@ -0,0 +1,458 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2014-present Datadog, Inc.
+
+//go:build linux
+
+package portlist
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"syscall"
+	"unsafe"
+
+	"go4.org/mem"
+	"golang.org/x/sys/unix"
+)
+
+func (p *Poller) init() {
+	p.os = newLinuxImpl(p.IncludeLocalhost)
+}
+
+type linuxImpl struct {
+	procNetFiles    []*os.File // seeked to start & reused between calls
+	readlinkPathBuf []byte
+
+	known            map[string]*portMeta // inode string => metadata
+	br               *bufio.Reader
+	includeLocalhost bool
+}
+
+type portMeta struct {
+	port          Port
+	pid           int
+	keep          bool
+	needsProcName bool
+}
+
+var eofReader = bytes.NewReader(nil)
+
+func newLinuxImplBase(includeLocalhost bool) *linuxImpl {
+	return &linuxImpl{
+		br:               bufio.NewReader(eofReader),
+		known:            map[string]*portMeta{},
+		includeLocalhost: includeLocalhost,
+	}
+}
+
+func newLinuxImpl(includeLocalhost bool) osImpl {
+	li := newLinuxImplBase(includeLocalhost)
+	for _, name := range []string{
+		"/proc/net/tcp",
+		"/proc/net/tcp6",
+		"/proc/net/udp",
+		"/proc/net/udp6",
+	} {
+		f, err := os.Open(name)
+		if err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+			log.Printf("portlist warning; ignoring: %v", err)
```

Comment:
Should it be a warning?

Reply:
All uses of `log` should probably be replaced by the agent's logger

Reply:
Done